gateleen-kafka/src/main/java/org/swisspush/gateleen/kafka/KafkaMessageSender.java

```diff
@@ -44,7 +63,45 @@ private Future<Void> sendMessage(KafkaProducer<String, String> kafkaProducer, Ka
         return kafkaProducer.send(message).compose((Function<RecordMetadata, Future<Void>>) metadata -> {
             log.debug("Message successfully sent to kafka topic '{}' on partition {} with offset {}. Timestamp: {}",
                     metadata.getTopic(), metadata.getPartition(), metadata.getOffset(), metadata.getTimestamp());
+            incrementSuccessCount(metadata.getTopic());
             return Future.succeededFuture();
-        }).onFailure(throwable -> log.warn("Failed to send message with key '{}' to kafka. Cause: {}", message.key(), throwable));
+        }).onFailure(throwable -> {
+            log.warn("Failed to send message with key '{}' to kafka. Cause: {}", message.key(), throwable);
+            incrementFailCount1(message.topic());
+        });
+    }
+
+    private void incrementSuccessCount(String topic) {
+        Counter counter = successSendCounterMap.get(topic);
+        if(counter != null) {
+            counter.increment();
+            return;
+        }
+
+        if(meterRegistry != null) {
+            Counter newCounter = Counter.builder(SUCCESS_SEND_MESSAGES_METRIC)
+                    .description(SUCCESS_SEND_MESSAGES_METRIC_DESCRIPTION)
+                    .tag(TOPIC, topic)
+                    .register(meterRegistry);
+            newCounter.increment();
+            successSendCounterMap.put(topic, newCounter);
+        }
+    }
```

Comment:
`incrementSuccessCount` and `incrementFailCount1` are NOT thread-save. Is it ensured only one thread is at work here (eg in case of multiple verticle instances)?